modules/toxiproxy/config.go

```diff
@@ -0,0 +1,78 @@
+package toxiproxy
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+)
+
+// proxy represents a single proxy configuration in the toxiproxy config file
+type proxy struct {
+	Name         string `json:"name"`
+	Listen       string `json:"listen"`
+	listenIP     string
+	listenPort   int
+	Upstream     string `json:"upstream"`
+	upstreamIP   string
+	upstreamPort int
+	Enabled      bool `json:"enabled"`
+}
+
+// sanitize is a helper function that returns another proxy
+// in which all the fields have been extracted from the
+// string representation of the upstream and listen fields.
+func (p proxy) sanitize() (proxy, error) {
```

Comment:
suggestion: act on a pointer so you can just return an error.
```suggestion
func (p *proxy) sanitize() error {
```